### ISSUE TITLE
Notes list features

### ISSUE THREAD

**Issue by dxerw** [pbek/QOwnNotes#526]
I love your project but I would like to recommend a few features if you don't mind :)

- An option to pin/favorite Notes and show them on top of everything else
- An option to show months (and the number of notes on the same month)
- An option to provide the title/date (either based on when it was last update for eg 20 minutes ago/yesterday/1 week ago/exact date if too old)/some of the text (if any) and a preview of an image (if there is one)

Something like this:
![](https://i.imgur.com/4AZ7cdV.png)
(Ehm my designing skills are non existent lol)

Thanks!

**Comment by pbek** [comment]
Thank you for your suggestions.
You can ask @Maboroshy about his script in #524 to achieve your first wish.

**Comment by Maboroshy** [comment]
@dxerw, what OS do you use?

**Comment by dxerw** [comment]
Thank you for the reply @pbek! I am currently using Windows 7 @Maboroshy mate :)

**Comment by Maboroshy** [comment]
Then I should put more effort into making that script windows compatible... :wink:

**Comment by Maboroshy** [comment]
Messing third evening with `script.startDetachedProcess` and `script.startSynchronousProcess` to rename files in Windows with no avail. :astonished: All I've got is some cmd windows with "invalid file path" flashing. And I thought making it work in Linux was pain...

I wonder if whole process scripting thing was tested in Windows. C++ code has some fine tuning for OSX but nothing for Windows. Not sure how it deals with "\" vs "/" issue.

**Comment by pbek** [comment]
Yes, maybe your paths have "/" in it? Qt uses "/" under Windows. (I learned that the hard and painful way)

**Comment by Maboroshy** [comment]
I've tried `fullFileName.replace("/","\\")` with no success.

Also in Windows you can't just do `move file1 file2`, it has no "move" binary, cause it's a cmd internal command. And that means my binary is cmd and `move file1 file2` is a **one** of it's arguments. And that means I must assemble it by myself with all the quoting hell it makes... And people think bash quoting is pain...

**Comment by pbek** [comment]
What if you would write a batch file that does the job and call that... (like I did for the Windows updater)

**Comment by Maboroshy** [comment]
I think batch file is an overkill for that simple task. I'll better wait till there'll be a full-featured "create note" scripting function and "delete note" function. :wink:
Windows stuff is pulling too much time, so I'll go with Linux-only for now. Maybe some "detach process guru" will make it work. It's Github after all.

**Comment by pbek** [comment]
When are you learning C++ and Qt? ;)

**Comment by Maboroshy** [comment]
When there'll be some functionality I really want that can be made only on C++. 
For now bash is enough... well, 1000 lines of bash. :grimacing:

**Comment by pbek** [comment]
Maybe you want to switch to Python? :)

**Comment by Maboroshy** [comment]
I really want to, even completed some online courses. But when I had free evening for hacking, which is not often these days, it was either starting something from the scratch on Python, with months till it become something cool, or putting more features to my script on bash. 

I guess I just don't have a good idea for some completely new project and have many ideas for Bash-note. I really use most of its features every day and with QOwnNotes scripting I can make it do almost anything as the back-end with QOwnNotes being the front-end.

**Comment by pbek** [comment]
That also sounds wonderful! :clap:

**Comment by Li-An** [comment]
+1 for all the suggestions here (Windows 10). The pin note feature is the one I loved in ResophNotes.
